lms/djangoapps/learner_dashboard/api/v0/views.py

```diff
@@ -0,0 +1,193 @@
+                            "uuid": "fe1a9ad4-a452-45cd-80e5-9babd3d43f96",
+                            "title": "Demonstration Course",
+                            "course_runs": [],
+                            "entitlements": [],
+                            "owners": [],
+                            "image": "",
+                            "short_description": "",
+                            "type": "457f07ec-a78f-45b4-ba09-5fb176520d8a",
+                        }
+                    ],
+                },
+                "certificate_data": [{
+                    "type": "course",
+                    "title": "edX Demo Course",
+                    'url': "/certificates/6e57d3cce8e34cfcb60bd8e8b04r07e0",
+                }],
+                "industry_pathways": [
+                    {
+                        "id": 2,
+                        "uuid": "1b8fadf1-f6aa-4282-94e3-325b922a027f",
+                        "name": "Demo Industry Pathway",
+                        "org_name": "edX",
+                        "email": "edx@edx.com",
+                        "description": "Sample demo industry pathway",
+                        "destination_url": "http://rit.edu/online/pathways/gtx-analytics-essential-tools-methods",
+                        "pathway_type": "industry",
+                        "program_uuids": [
+                            "a156a6e2-de91-4ce7-947a-888943e6b12a"
+                        ]
+                    }
+                ],
+                "credit_pathways": [
+                    {
+                        "id": 1,
+                        "uuid": "86b9701a-61e6-48a2-92eb-70a824521c1f",
+                        "name": "Demo Credit Pathway",
+                        "org_name": "edX",
+                        "email": "edx@edx.com",
+                        "description": "Sample demo credit pathway!",
+                        "destination_url": "http://rit.edu/online/pathways/ritx-design-thinking",
+                        "pathway_type": "credit",
+                        "program_uuids": [
+                            "a156a6e2-de91-4ce7-947a-888943e6b12a"
+                        ]
+                    }
+                ]
+            }
+    """
+
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+
+    permission_classes = (permissions.JWT_RESTRICTED_APPLICATION_OR_USER_ACCESS,)
+
+    def get(self, request, program_uuid):
+        """
+        Retrieves progress details of a user in a specified program.
+
+        Args:
+            request (Request): Django request object.
+            program_uuid (string): URI element specifying uuid of the program.
+
+        Return:
+        """
+        programs_config = ProgramsApiConfig.current()
+        user = request.user
+        site = request.site
+        if not programs_config.enabled:
```

Comment:
do we really need this check?

Reply:
Just checked this configuration model, it's only there to set up a few program related configurations. Mostly used to construct program's marketing url which in this cases is not needed. Working on discarding this check.

Reply:
Done